be/src/exprs/vectorized/array_functions.tpp

```diff
@@ -105,6 +105,303 @@ private:
+        DCHECK_EQ(columns.size(), 1);
+
+        size_t chunk_size = columns[0]->size();
+        ColumnPtr src_column = ColumnHelper::unpack_and_duplicate_const_column(chunk_size, columns[0]);
+        ColumnPtr dest_column = src_column->clone_empty();
+
+        if (columns[0]->is_nullable()) {
+            const auto* src_nullable_column = down_cast<const NullableColumn*>(src_column.get());
+            const auto* src_data_column = down_cast<const ArrayColumn*>(src_nullable_column->data_column().get());
+            auto& dest_nullable_column = down_cast<NullableColumn&>(*dest_column);
+            auto& dest_null_data = down_cast<NullableColumn&>(*dest_column).null_column_data();
+            auto& dest_data_column = down_cast<ArrayColumn&>(*dest_nullable_column.data_column());
+
+            dest_null_data = src_nullable_column->immutable_null_column_data();
+            dest_nullable_column.set_has_null(src_nullable_column->has_null());
+
+            if (src_nullable_column->has_null()) {
+                for (size_t i = 0; i < chunk_size; i++) {
+                    if (!src_nullable_column->is_null(i)) {
+                        _array_difference_item(*src_data_column, i, &dest_data_column);
+                    } else {
+                        dest_data_column.append_default();
+                    }
+                }
+            } else {
+                for (size_t i = 0; i < chunk_size; i++) {
+                    _array_difference_item(*src_data_column, i, &dest_data_column);
+                }
+            }
+        } else {
+            const auto* src_data_column = down_cast<const ArrayColumn*>(src_column.get());
+            auto* dest_data_column = down_cast<ArrayColumn*>(dest_column.get());
+
+            for (size_t i = 0; i < chunk_size; i++) {
+                _array_difference_item(*src_data_column, i, dest_data_column);
+            }
+        }
+        return dest_column;
+    }
+
+    static void _array_difference_item(const ArrayColumn& column, size_t index, ArrayColumn* dest_column) {
+        Datum v = column.get(index);
+        const auto& items = v.get<DatumArray>();
+
+        auto& dest_data_column = dest_column->elements_column();
+        for (size_t i = 0; i < items.size(); ++i) {
+            if (i == 0) {
+                if (items[i].is_null()) {
+                    dest_data_column->append_nulls(1);
+                } else {
+                    dest_data_column->append_datum((CppType)0);
+                }
+            } else {
+                if (items[i - 1].is_null() || items[i].is_null()) {
+                    dest_data_column->append_nulls(1);
+                } else {
+                    dest_data_column->append_datum((CppType)(items[i].get<CppType>() - items[i - 1].get<CppType>()));
+                }
+            }
+        }
+
+        auto& dest_offsets = dest_column->offsets_column()->get_data();
+        dest_offsets.emplace_back(dest_offsets.back() + items.size());
```

Comment:
can offsets directed copy from src offsets ?

Reply:
> can offsets directed copy from src offsets ?

Yes, let me improve this.